### DISCUSSION TITLE
Can we Invalidate mutation so that we get updated data

### DISCUSSION THREAD

**Discussion by noumanaltaf** [TanStack/query#6791] [Q&A]
Can we invalidate mutation query data for some specific key to get updated data on UI where useMutation data is used?
here is a code example to show what I want to achieve

`  const queryClient = useQueryClient();

    const calculateMutation = useMutation(calculateFunction)
    const { data, mutate } = useMutation(updateFuntion, {
        onSuccess: (data) => {
            queryClient.setQueryData(['updateQueryData'], data)
        }
    });

    const update = React.useCallback(
        () => {
            mutate({ a: 1 });
        },
        []
    );
    
    const onCalculate = React.useCallback(
        () => {
            const response = calculateMutation.mutateAsync({ a: 1 });
            queryClient.setQueryData(['updateQueryData'], response.data)
        },
        []
    );

    return (
        <Component
            componentData={data}
            onCalculate={onCalculate}
        />
    )

`

So in the above code, when I update '**updateQueryData**' data. The component is not getting updated data in **componentData** prop. 
Is there any way to get updated data to pass into componentData prop? I don't want to use component useState our here.

**Comment by TkDodo** [comment]
you're updating a query, not a mutation. you can't update a mutation
